src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6190,67 +6216,63 @@ void VM_Crac::doit() {
       const int mjr = major(st->st_rdev);
       const int mnr = minor(st->st_rdev);
       if (mjr == 1 && (mnr == 8 || mnr == 9)) {
-        if (CRPrintResourcesOnCheckpoint) {
-          tty->print_cr("OK: always available, random or urandom");
-        }
+        print_resources(ostream, "OK: always available, random or urandom");
         continue;
       }
     }
 
     if (fds.check(i, FdsInfo::M_CLASSPATH) && !fds.check(i, FdsInfo::M_CANT_RESTORE)) {
-      if (CRPrintResourcesOnCheckpoint) {
-        tty->print_cr("OK: in classpath");
-      }
+      print_resources(ostream, "OK: in classpath");
       continue;
     }
 
     if (fds.check(i, FdsInfo::M_PERSISTENT)) {
-      if (CRPrintResourcesOnCheckpoint) {
-        tty->print_cr("OK: assured persistent");
-      }
+      print_resources(ostream, "OK: assured persistent");
       continue;
     }
 
-    if (CRPrintResourcesOnCheckpoint) {
-      tty->print("BAD: opened by application");
-    }
-    ok = false;
-
     if (S_ISSOCK(st->st_mode)) {
-      details = sock_details(details, detailsbuf, sizeof(detailsbuf));
-      if (CRPrintResourcesOnCheckpoint) {
-        tty->print(" details2=\"%s\" ", details);
+      if (is_socket_from_jcmd(i)){
+        print_resources(ostream, "OK: jcmd socket");
+        ok = true;
+        continue;
       }
+      details = sock_details(details, detailsbuf, sizeof(detailsbuf));
+      print_resources(ostream, "issock, details2=\"%s\" ", details);
     }
 
-    if (CRPrintResourcesOnCheckpoint) {
-      tty->cr();
-    }
+    print_resources(ostream, "BAD: opened by application");
+    ok = false;
+
     char* msg = NEW_C_HEAP_ARRAY(char, strlen(details) + 1, mtInternal);
     strcpy(msg, details);
     _failures->append(CracFailDep(stat2stfail(st->st_mode & S_IFMT), msg));
   }
 
-  if (!ok && CRHeapDumpOnCheckpointException) {
-    HeapDumper::dump_heap();
-  }
-
-  if (!ok && CRDoThrowCheckpointException) {
+  if (!ok) {
+    if (CRHeapDumpOnCheckpointException){
+      HeapDumper::dump_heap();
+    }
```

Comment:
What is the reason for this change? CRDoThrowCheckpointException is useful sometimes. Please revert back.